File: src/cli/table.test.ts
import { describe, expect, test } from "bun:test";

import { createQueryPlan } from "@/cli/plan";
import { renderTable } from "@/cli/table";
import type { SearchResult } from "@/shared/types";

describe("table rendering", () => {
  test("renders compact availability output", () => {
    const results: SearchResult[] = [
      {
        domain: "murk.ink",
        available: true,
        price: "2.00",
        currency: "USD",
        buyUrl: "https://example.com/?d=murk.ink",
        source: "dynadot",
        cached: false,
        quotedAt: "2026-02-20T00:00:00.000Z",
      },
    ];

    const plan = createQueryPlan(["murk.ink"]);

    const table = renderTable(results, plan.groups);

    expect(table).toContain(" · ");
    expect(table).toContain("murk.ink");
    expect(table).toContain("2.00");
  });

  test("renders suggestion tree with connectors", () => {
    const results: SearchResult[] = [
      {
        domain: "murk.com",
        available: false,
        price: null,
        currency: "USD",
        buyUrl: null,
        source: "dynadot",
        cached: false,
        quotedAt: "2026-02-20T00:00:00.000Z",
      },
      {
        domain: "murk.sh",
        available: true,
        price: "39.99",
        currency: "USD",
        buyUrl: "https://example.com/?d=murk.sh",
        source: "dynadot",
        cached: false,
        quotedAt: "2026-02-20T00:00:00.000Z",
      },
    ];

    const plan = createQueryPlan(["murk"]);
    const output = renderTable(results, plan.groups);

    expect(output).toContain("murk");
    expect(output).toContain("├─");
    expect(output).toContain("└─");
    expect(output).toContain("Taken");
  });
});

File: internal/cli/table_test.go
package cli

import (
	"os"
	"strings"
	"testing"

	"dotld/internal/shared"
)

func TestMain(m *testing.M) {
	os.Setenv("NO_COLOR", "1")
	useColor = false
	os.Exit(m.Run())
}

func TestAvailableOutput(t *testing.T) {
	price := "2.00"
	buyURL := "https://example.com/?d=murk.ink"
	results := []shared.SearchResult{
		{
			Domain:    "murk.ink",
			Available: true,
			Price:     &price,
			Currency:  "USD",
			BuyURL:    &buyURL,
			Cached:    false,
			QuotedAt:  "2026-02-20T00:00:00.000Z",
		},
	}

	groups, _ := createQueryPlan([]string{"murk.ink"})
	table := renderTable(results, groups)

	if !strings.Contains(table, " · ") {
		t.Error("expected divider in output")
	}
	if !strings.Contains(table, "murk.ink") {
		t.Error("expected domain in output")
	}
	if !strings.Contains(table, "2.00") {
		t.Error("expected price in output")
	}
}

func TestSuggestionTreeConnectors(t *testing.T) {
	price := "39.99"
	buyURL := "https://example.com/?d=murk.sh"
	results := []shared.SearchResult{
		{
			Domain:    "murk.com",
			Available: false,
			Price:     nil,
			Currency:  "USD",
			BuyURL:    nil,
			Cached:    false,
			QuotedAt:  "2026-02-20T00:00:00.000Z",
		},
		{
			Domain:    "murk.sh",
			Available: true,
			Price:     &price,
			Currency:  "USD",
			BuyURL:    &buyURL,
			Cached:    false,
			QuotedAt:  "2026-02-20T00:00:00.000Z",
		},
	}

	groups, _ := createQueryPlan([]string{"murk"})
	output := renderTable(results, groups)

	if !strings.Contains(output, "murk") {
		t.Error("expected root label in output")
	}
	if !strings.Contains(output, "├─") {
		t.Error("expected tree connector ├─")
	}
	if !strings.Contains(output, "└─") {
		t.Error("expected tree connector └─")
	}
	if !strings.Contains(output, "Taken") {
		t.Error("expected Taken in output")
	}
}
